src/frontend/src/eth/utils/qr-code.utils.ts

```diff
@@ -19,13 +23,58 @@ export const decodeQrCode = ({
 		return { status: 'cancelled' };
 	}
 
-	const payment = decodePayment(code);
+	const payment = decodeQrCodeUrn(code);
 
 	if (isNullish(payment)) {
 		return { status: 'success', destination: code };
 	}
 
-	const { token, identifier: destination, amount } = payment;
+	const { prefix, destination, amount, address, ethereumChainId, functionName } = payment;
+
+	const normalizeChainId = (chainId: string): string => {
+		if (chainId.startsWith('0x')) {
```

Comment:
do we have maybe already a utils for this?

Reply:
you mean to parse a string to int right?

I found [`hexStringToUint8Array`](https://github.com/dfinity/ic-js/blob/0e762d9405c1f58ccfae2af6e73cab5214ed44d0/packages/utils/src/utils/arrays.utils.ts#L52) in `@dfinity/utils`.

Or you mean a `normalizeChainId` util?

Reply:
I mean `normalizeChainId` and `startsWith('0x')`

Reply:
Not that I know of. I searched in both `@dfinity/utils` and `ethers.js` but I could not find something similar. Shall I create it in our utils?

Reply:
It would be nice I think, no?

It can be extracted after we merge this PR.

Reply:
will do!